src/backend/distributed/commands/aggregate.c

```diff
@@ -0,0 +1,62 @@
+/*-------------------------------------------------------------------------
+ *
+ * aggregate.c
+ *    Commands for distributing AGGREGATE statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/commands.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/deparser.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/metadata/distobject.h"
+#include "nodes/parsenodes.h"
+
+/*
+ * PreprocessDefineAggregateStmt is called during the planning phase for
+ * CREATE AGGREGATE ..
+ */
+List *
+PreprocessDefineAggregateStmt(Node *node, const char *queryString,
+							  ProcessUtilityContext processUtilityContext)
+{
+	QualifyTreeNode((Node *) node);
+
+	return NIL;
+}
+
+
+List *
+PostprocessDefineAggregateStmt(Node *node, const char *queryString)
+{
+	DefineStmt *stmt = castNode(DefineStmt, node);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
+
+	if (IsObjectAddressOwnedByExtension(&address, NULL))
+	{
+		return NIL;
+	}
+
```

Comment:
We should check whether there is any undistributable dependent object and if so we should create aggregate locally (Just like functions)